o1vm/src/pickles/proof.rs

```diff
@@ -1,15 +1,17 @@
 use kimchi::curve::KimchiCurve;
 use poly_commitment::{ipa::OpeningProof, PolyComm};
 
-pub struct WitnessColumns<G> {
+use crate::interpreters::mips::column::N_MIPS_SEL_COLS;
+
+pub struct WitnessColumns<G, S> {
     pub scratch: [G; crate::interpreters::mips::witness::SCRATCH_SIZE],
     pub instruction_counter: G,
     pub error: G,
-    pub selector: G,
+    pub selector: S,
```

Comment:
I'm not a big fan of this, but I do not have a better idea for now. Maybe `Vec<G>`? It would `ScalarField` when we do have the single value for the row, and `Vec<ScalarField>` when we deal with the evaluations of all selectors? The only thing is that the number of selectors would be "unknown". Which, I think, is fine.

Reply:
Changing this to `Vec<G>` would, I believe, make it impossible to fix the previous comment, as then `G` would need to be `usize`.

Reply:
You can use `Fp::from(x as usize)`.